src/Lean/Elab/DocString/Builtin.lean

```diff
@@ -35,6 +35,16 @@ set_option linter.missingDocs true
 
 public section
 
+/--
+As elaboration results are not added to the environment as part of constants, manually record
+references so that `shake` keeps their imports.
+-/
+private def elabExtraTerm (stx : Syntax) (expectedType? : Option Expr := none) : TermElabM Expr := do
+  let e ← elabTermAndSynthesize stx expectedType?
```

Comment:
@david-christiansen It looks like I broke a test by adding `AndSynthesize` here, was the test skipping to find an instance intentional? Without this, I was missing some references to instances.